x-pack/plugins/siem/public/management/common/routing.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { generatePath } from 'react-router-dom';
+import {
+  MANAGEMENT_ROUTING_ENDPOINTS_PATH,
+  MANAGEMENT_ROUTING_POLICIES_PATH,
+  MANAGEMENT_ROUTING_POLICY_DETAILS_PATH,
+  MANAGEMENT_ROUTING_ROOT_PATH,
+} from './constants';
+import { ManagementSubTab } from '../types';
+import { SiemPageName } from '../../app/types';
+
+export type getManagementUrlArgs =
+  | ['endpointList']
+  | ['policyList']
+  | ['policyDetails', { policyId: string }];
+
+/**
+ * Returns a URL string for a given Management page view
+ * @param options
+ */
+export const getManagementUrl = (...args: getManagementUrlArgs): string => {
```

Comment:
The purpose of this method is to build URLs using the React Router paths.

Question: is there already a hook/lib/util that centralizes the building up URLs across SIEM?

One issue about this method here is that it does not know the "root" of the URL (everything to the left of the `#` (including the hash itself).